interface/controller/app_controller.go

```diff
@@ -1,5 +1,73 @@
 package controller
```

Comment:
Is redundant to name the file app_controller, as you are inside the controller package. Please consider fixing this naming redundancy here and all other files in your project.